src/map/skill.c

```diff
@@ -11027,17 +11028,18 @@ static int skill_castend_map(struct map_session_data *sd, uint16 skill_id, const
 		return 0;
 	}
 
-	switch(skill_id) {
+	switch (skill_id) {
 		case AL_TELEPORT:
-			// The storage window is closed automatically by the client when there's
-			// any kind of map change, so we need to restore it automatically
-			// issue: 8027
-			if(strcmp(mapname,"Random")==0)
-				pc->randomwarp(sd,CLR_TELEPORT);
-			else if (sd->menuskill_val > 1) //Need lv2 to be able to warp here.
-				pc->setpos(sd,sd->status.save_point.map,sd->status.save_point.x,sd->status.save_point.y,CLR_TELEPORT);
-
-			clif->refresh_storagewindow(sd);
+			if (strcmp(mapname, "Random") == 0)
+				pc->randomwarp(sd, CLR_TELEPORT);
+			else if (sd->menuskill_val > 1) // Need lv2 to be able to warp here.
+				pc->setpos(sd, sd->status.save_point.map, sd->status.save_point.x, sd->status.save_point.y, CLR_TELEPORT);
+
+			if (battle_config.teleport_close_storage == 1 && sd->state.storage_flag != STORAGE_FLAG_CLOSED) {
+				clif->storageclose(sd);
+				sd->state.storage_flag = STORAGE_FLAG_CLOSED;
+			} else
+				clif->refresh_storagewindow(sd);
```

Comment:
From [style guide](https://github.com/HerculesWS/Hercules/wiki/Coding-Style#chapter-3-placing-braces-and-spaces):

> Do not unnecessarily use braces where a single statement will do.
> This does not apply if only one branch of a conditional statement is a single statement; in the latter case use braces in both branches.